ATLAS-5002 : Support Java 17 for build and runtime

pareshddevalia opened the pull request:
## What changes were proposed in this pull request?

This PR introduces support for Java 17 across both build and runtime environments, while retaining compatibility with Java 8. The system can now be built and executed seamlessly using either JDK 8 or JDK 17.

## How was this patch tested?

The patch was validated on both JDK 8 and JDK 17 through full Maven build cycles and runtime execution.

Maven Build Verification

JDK 8:
mvn clean install

JDK 17:
mvn clean install

Scenario 1: JDK 8 Build and Runtime
Scenario 2: JDK 8 Build and JDK 17 Runtime
Scenario 3: JDK 17 Build and Runtime

Note for JDK 17 Users:
Running integration tests (especially Jetty server-based tests) with JDK 17 requires additional JVM arguments due to stricter module access. These can be provided via MAVEN_OPTS:

```
export MAVEN_OPTS="--add-opens=java.base/java.lang=ALL-UNNAMED \
--add-opens=java.base/java.lang.reflect=ALL-UNNAMED \
--add-opens=java.base/java.nio=ALL-UNNAMED \
--add-opens=java.base/java.net=ALL-UNNAMED "
```

Alternatively, you can place these arguments in a .mvn/jvm.config file to avoid exporting them manually for every run.

pareshD-1991 committed:
ATLAS-5002 : Support Java 17 for build and runtime

# Conflicts:
#	.github/workflows/ci.yml
#	notification/src/test/java/org/apache/atlas/notification/RestNotificationTest.java

mneethiraj commented on webapp/src/test/resources/template_metadata.csv:
It looks like tests update this file; please review if changes to this file need to be included in this commit.

pareshD-1991 committed:
Test update this .csv file

pareshD-1991 committed:
Fix the python code, removed unnecessary JPMS and increase the stop wait sec

pareshD-1991 committed:
updated the Readme file

UmeshPatil-1 committed:
Refactor deprecated Mockito Matchers to ArgumentMatchers and fix failing Hive test cases

UmeshPatil-1 committed:
ATLAS-5002: Refactor Mockito imports: remove wildcard and deprecated Matchers usage

UmeshPatil-1 committed:
ATLAS-5002: Refactor Mockito imports

pareshD-1991 committed:
Removed hardcoded m2 path from docker

UmeshPatil-1 committed:
ATLAS-5002: Resolved Reflection error and refactor mockito imports.

pareshD-1991 committed:
resolved conflict with latest rebase master branch

prabhjyotsingh commented:
Any updates on this, are we getting this?  
:)

pareshddevalia commented:
> Any updates on this, are we getting this? :)

Yes, we are in process of review state.

pareshD-1991 committed:
ATLAS-5002 : Added support for openapi-3.0 for API documentation and fix some test-case.

pareshD-1991 committed:
ATLAS-5002: Bundle commons-text in hbase-hook

1) commons-configuration2 in coprocessor without commons-text.jar on plugin CP
java.lang.NoClassDefFoundError: org/apache/commons/text/lookup/StringLookupFactory
	at org.apache.commons.configuration2.interpol.DefaultLookups.<clinit>(...)

Caused by: java.lang.ClassNotFoundException: org.apache.commons.text.lookup.StringLookupFactory
	at org.apache.atlas.plugin.classloader.AtlasPluginClassLoader$MyClassLoader.findClass(...)

Also surfaced as:
The coprocessor org.apache.atlas.hbase.hook.HBaseAtlasCoprocessor threw
java.lang.InstantiationException: org.apache.atlas.hbase.hook.HBaseAtlasCoprocessor

pareshD-1991 committed:
ATLAS-5002 : Fix new test-case for Repository module.
